modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/source/poll/PollingSourceWrapper.java

```diff
@@ -754,6 +797,29 @@ private void shutdownScheduler() {
     }
   }
 
+  private int statusToNotificationType(PollContext.PollItemStatus status, boolean currentPollItemLimitApplied) {
+    if (currentPollItemLimitApplied) {
+      return ITEM_REJECTED_LIMIT;
+    }
+    switch (status) {
+      case ACCEPTED:
+        return ITEM_DISPATCHED;
+      case FILTERED_BY_WATERMARK:
+        return ITEM_REJECTED_WATERMARK;
+      case ALREADY_IN_PROCESS:
+        return ITEM_REJECTED_IDEMPOTENCY;
+      case SOURCE_STOPPING:
+        return ITEM_REJECTED_SOURCE_STOPPING;
+    }
+    return NO_ACTION_ID;
+  }
+
+  private void dispatchPollingSourceNotification(int action, String componentLocation, String pollId) {
```

Comment:
there was a comment about deleting this method. Where did it go?

Reply:
I reworked the method to avoid the points you raised, maybe to the refactor the comment went away? Anyways please look at it again